draft-ietf-tls-esni.md

```diff
@@ -429,7 +429,15 @@ may use the "ech_outer_extensions" extension.
 ~~~
 
 OuterExtensions consists of one or more ExtensionType values, each of which
-reference an extension in ClientHelloOuter.
+reference an extension in ClientHelloOuter. The extensions in OuterExtensions
+MUST appear in ClientHelloOuter in the same relative order, however, there is
+no requirement that they be continguous. For examples, OuterExtensions may
+contain extensions A, B, C, while ClientHelloOuter contains extensions A, D, B,
+C, E, F.
+
+The "ech_outer_extensions" extension is only used for compressing the
+ClientHelloInner. It MUST NOT be sent in either ClientHelloOuter or
+ClientHelloInner.
```

Comment:
```suggestion
ClientHelloInner. It can only be included in EncodedClientHelloInner, and MUST
NOT be sent in either ClientHelloOuter or ClientHelloInner.
```

Reply:
I was really confused by this at first, so hopefully this helps clarify for others.